chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1748,106 @@ def update_lnet_configuration(self, lnet_configuration_list):
     @property
     def CommandPlan(self):
         return CommandPlan(self._lock_cache, self._job_collection)
+
+    def configure_stratagem(self, stratagem_data):
+        with self._lock:
+            configuration_data = {
+                "state": "unconfigured",
+                "filesystem_id": stratagem_data.get("filesystem"),
+                "interval": stratagem_data.get("interval"),
+                "report_duration": stratagem_data.get("report_duration"),
+                "report_duration_active": stratagem_data.get("report_duration_active"),
+                "purge_duration": stratagem_data.get("purge_duration"),
+                "purge_duration_active": stratagem_data.get("purge_duration_active"),
+            }
+
+            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
+            fs_identifier = str(configuration_data.get("filesystem_id"))
```

Comment:
What happens if the dict doesn't have this value?

Reply:
In other words:

```
>>> str(None)
'None'
```

Reply:
Actually, This and the fs lookup below could be replaced by using `Q`.

```python
fs_id = configuration_data.get("filesystem_id")
fs = ManagedFilesystem.objects.filter(Q(id=fs_id) | Q(name=fs_id)).first()
# Check if fs is None here and return error?
configuration_data["filesystem_id"] = fs.id
```

Reply:
This also has the advantage of not returning the whole table, and the SQL generated is not bad:

```sql
SELECT "chroma_core_managedfilesystem"."id" FROM "chroma_core_managedfilesystem" WHERE ("chroma_core_managedfilesystem"."not_deleted" = True  AND ("chroma_core_managedfilesystem"."id" = 3  OR "chroma_core_managedfilesystem"."name" = fs )) ORDER BY "chroma_core_managedfilesystem"."id" ASC
```

Reply:
This is now using the `get_fs_id_from_identifier` so if fs_identifier was `None` it would simply return `None` for fs_id and the function would return. Here is output from a test I ran:

```
69 	            # The filesystem_id may come in as the fs name or the fs id. In terms of storing information in the database, the fs id should always be used.
1770 	            fs_identifier = str(configuration_data.get("filesystem_id"))
1771 	            from remote_pdb import RemotePdb
1772 	            RemotePdb('127.0.0.1', 4444).set_trace()
1773 ->	            fs_id = get_fs_id_from_identifier(fs_identifier)
1774 	
1775 	            if not fs_id:
1776 	                return None
1777 	
1778 	            configuration_data["filesystem_id"] = fs_id
(Pdb) fs_identifier
'fs23'
(Pdb) fs_identifier = None
(Pdb) fs_identifier
(Pdb) n
> /usr/share/chroma-manager/chroma_core/services/job_scheduler/job_scheduler.py(1775)configure_stratagem()
-> if not fs_id:
(Pdb) fs_id
```

However, i'm not sure that this scenario is possible as the validator would ensure the filesystem exists before creating the bundle data and invoking the JobSchedulerClient.configure_stratagem function.

Reply:
Will raise error.